src/data/transforms/minmax.rs

```diff
@@ -0,0 +1,247 @@
+//! The Min-Max transformer
+//!
+//! This module contains the `MinMaxScaler` transformer.
+//!
+//! The `MinMaxScaler` transformer is used to transform input data
+//! so that the minimum and maximum of each column are as specified.
+//! This is commonly used to transform the data to have a minimum of
+//! `0` and a maximum of `1`.
+//!
+//! # Examples
+//!
+//! ```
+//! use rusty_machine::data::transforms::{Transformer, MinMaxScaler};
+//! use rusty_machine::linalg::Matrix;
+//!
+//! // Constructs a new `MinMaxScaler` to map minimum to 0 and maximum
+//! // to 1.
+//! let mut transformer = MinMaxScaler::default();
+//!
+//! let inputs = Matrix::new(2, 2, vec![-1.0, 2.0, 1.5, 3.0]);
+//!
+//! // Transform the inputs to get output data with required minimum
+//! // and maximum.
+//! let transformed = transformer.transform(inputs).unwrap();
+//! ```
+
+use learning::error::{Error, ErrorKind};
+use linalg::Matrix;
+use super::Transformer;
+
+use libnum::Float;
+
+/// The `MinMaxScaler`
+///
+/// The `MinMaxScaler` provides an implementation of `Transformer`
+/// which allows us to transform the input data to have a new minimum
+/// and maximum per column.
+///
+/// See the module description for more information.
+#[derive(Debug)]
+pub struct MinMaxScaler<T: Float> {
+    /// Values to scale each column by
+    scale_factors: Option<Vec<T>>,
+    /// Values to add to each column after scaling
+    const_factors: Option<Vec<T>>,
+    /// The min of the new data (default 0)
+    scaled_min: T,
+    /// The max of the new data (default 1)
+    scaled_max: T,
+}
+
+/// Create a default `MinMaxScaler` with minimum of `0` and
+/// maximum of `1`.
+impl<T: Float> Default for MinMaxScaler<T> {
+    fn default() -> MinMaxScaler<T> {
+        MinMaxScaler {
```

Comment:
I'm not sure I understand. There is a `new` function below this one - this is an implementation of the `Default` trait. I think it makes sense to have this here as I think scaling between 0 and 1 is common enough to warrant it as a default.